src/main/java/io/reactivex/internal/operators/flowable/FlowableDoOnEach.java

```diff
@@ -282,11 +305,33 @@ public int requestFusion(int mode) {
         @Nullable
         @Override
         public T poll() throws Exception {
-            T v = qs.poll();
+            T v;
+
+            try {
+                v = qs.poll();
+            } catch (Throwable ex) {
+                Exceptions.throwIfFatal(ex);
+                try {
+                    onError.accept(ex);
+                } catch (Throwable exc) {
+                    throw new CompositeException(ex, exc);
+                }
+                throw ExceptionHelper.<Exception>throwIfThrowable(ex);
```

Comment:
Hmm I don't understand why we both signal `onError` and also re-throw. It seems like it should be one or the other, no?